components/clp-package-utils/clp_package_utils/general.py

```
@@ -132,59 +130,59 @@ def generate_container_name(job_type: str) -> str:
+
+    :param project_name:
+    :return: Whether at least one instance is running.
+    :raises EnvironmentError: If Docker Compose is not installed or fails.
+    """
+    cmd = ["docker", "compose", "ls", "--format", "json", "--filter", f"name={project_name}"]
+    try:
+        output = subprocess.check_output(cmd, stderr=subprocess.STDOUT)
+        running_instances = json.loads(output)
+        return len(running_instances) >= 1
+    except subprocess.CalledProcessError:
+        raise EnvironmentError("docker-compose is not installed or not functioning properly.")
```

Ruff (0.14.0)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)